Unable to compile this repository on Windows strikes back

Recent scheduled CI jobs of the superbuild started failing due on Windows, see https://github.com/robotology/robotology-superbuild/actions/runs/56405001 . This is probably due to https://github.com/robotology/walking-controllers/pull/55, as at the moment the superbuild uses the master branch of walking-controllers both for stable and unstable branches. 

The error in particular seems to be:
~~~
2020-03-16T02:39:46.9396578Z LINK : fatal error LNK1104: cannot open file '..\YarpUtilities\Debug\WalkingControllersYarpUtilities.lib' [D:\a\robotology-superbuild\robotology-superbuild\build\robotology\walking-controllers\src\LoggerClient\LoggerClient.vcxproj] [D:\a\robotology-superbuild\robotology-superbuild\build\walking-controllers.vcxproj]
2020-03-16T02:39:47.0120257Z     Building Custom Rule D:/a/robotology-superbuild/robotology-superbuild/robotology/walking-controllers/src/RetargetingHelper/CMakeLists.txt
2020-03-16T02:39:47.0925924Z     Helper.cpp
2020-03-16T02:39:48.1668559Z D:\a\robotology-superbuild\robotology-superbuild\build\install\include\iDynTree/Core/EigenHelpers.h(275,14): warning C4244: 'argument': conversion from 'const ptrdiff_t' to 'const unsigned int', possible loss of data [D:\a\robotology-superbuild\robotology-superbuild\build\robotology\walking-controllers\src\RetargetingHelper\RetargetingHelper.vcxproj] [D:\a\robotology-superbuild\robotology-superbuild\build\walking-controllers.vcxproj]
2020-03-16T02:39:48.7331865Z LINK : fatal error LNK1104: cannot open file '..\YarpUtilities\Debug\WalkingControllersYarpUtilities.lib' [D:\a\robotology-superbuild\robotology-superbuild\build\robotology\walking-controllers\src\RetargetingHelper\RetargetingHelper.vcxproj] [D:\a\robotology-superbuild\robotology-superbuild\build\walking-controllers.vcxproj]
~~~

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,7 +5,7 @@ All notable changes to this project are documented in this file.
 ### Added
 
 ### Changed
-
+- Fix the windows compilation (https://github.com/robotology/walking-controllers/pull/59)
 ## [0.3.0] - 2020-03-16
 ### Added
 - The `CHANGELOG.md` file

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ set(CMAKE_CXX_STANDARD 14)
 
 ## MAIN project
 project(WalkingControllers
-  VERSION 0.3.0)
+  VERSION 0.3.1)
 
 # Defines the CMAKE_INSTALL_LIBDIR, CMAKE_INSTALL_BINDIR and many other useful macros.
 # See https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,34 @@ set(CMAKE_CXX_STANDARD 14)
 project(WalkingControllers
   VERSION 0.3.0)
 
+# Defines the CMAKE_INSTALL_LIBDIR, CMAKE_INSTALL_BINDIR and many other useful macros.
+# See https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html
+include(GNUInstallDirs)
+
+# Control where libraries and executables are placed during the build.
+# With the following settings executables are placed in <the top level of the
+# build tree>/bin and libraries/archives in <top level of the build tree>/lib.
+# This is particularly useful to run ctests on libraries built on Windows
+# machines: tests, which are executables, are placed in the same folders of
+# dlls, which are treated as executables as well, so that they can properly
+# find the libraries to run. This is a because of missing RPATH on Windows.
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+
+# To build shared libraries in Windows, we set CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS to TRUE.
+# See https://cmake.org/cmake/help/v3.4/variable/CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS.html
+# See https://blog.kitware.com/create-dlls-on-windows-without-declspec-using-new-cmake-export-all-feature/
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+# Under MSVC, we set CMAKE_DEBUG_POSTFIX to "d" to add a trailing "d" to library
+# built in debug mode. In this Windows user can compile, build and install the
+# library in both Release and Debug configuration avoiding naming clashes in the
+# installation directories.
+if(MSVC)
+    set(CMAKE_DEBUG_POSTFIX "d")
+endif()
+
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 # Check WalkingControllers dependencies, find necessary libraries.

